Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory.go

```diff
@@ -108,11 +110,15 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
 	flags.SetNormalizeFunc(util.WarnWordSepNormalizeFunc) // Warn for "_" flags
 
-	generators := map[string]kubectl.Generator{
-		"run/v1":     kubectl.BasicReplicationController{},
-		"run-pod/v1": kubectl.BasicPod{},
-		"service/v1": kubectl.ServiceGeneratorV1{},
-		"service/v2": kubectl.ServiceGeneratorV2{},
+	generators := map[string]map[string]kubectl.Generator{
```

Comment:
If we're at the point of enumerating which generators are valid for which commands, does it start to make more sense to have a structured set of input arguments that are plumbed through?  This is essentially saying callers have to be aware of the shape of magic container they are sending in.